virtualizarr/tests/test_xarray.py

```diff
@@ -310,27 +310,54 @@ def test_anon_read_s3(self, filetype, indexes):
 
 @network
 class TestReadFromURL:
-    def test_read_from_url(self):
-        examples = {
-            "grib": "https://github.com/pydata/xarray-data/raw/master/era5-2mt-2019-03-uk.grib",
-            "netcdf3": "https://github.com/pydata/xarray-data/raw/master/air_temperature.nc",
-            "netcdf4": "https://github.com/pydata/xarray-data/raw/master/ROMS_example.nc",
-            "hdf4": "https://github.com/corteva/rioxarray/raw/master/test/test_data/input/MOD09GA.A2008296.h14v17.006.2015181011753.hdf",
+    @pytest.mark.parametrize(
+        "filetype, url",
+        [
+            (
+                "grib",
+                "https://github.com/pydata/xarray-data/raw/master/era5-2mt-2019-03-uk.grib",
+            ),
+            (
+                "netcdf3",
+                "https://github.com/pydata/xarray-data/raw/master/air_temperature.nc",
+            ),
+            (
+                "netcdf4",
+                "https://github.com/pydata/xarray-data/raw/master/ROMS_example.nc",
+            ),
+            (
+                "hdf4",
+                "https://github.com/corteva/rioxarray/raw/master/test/test_data/input/MOD09GA.A2008296.h14v17.006.2015181011753.hdf",
+            ),
             # https://github.com/zarr-developers/VirtualiZarr/issues/159
-            # "hdf5": "https://github.com/fsspec/kerchunk/raw/main/kerchunk/tests/NEONDSTowerTemperatureData.hdf5",
-            # https://github.com/zarr-developers/VirtualiZarr/issues/160
-            # "tiff": "https://github.com/fsspec/kerchunk/raw/main/kerchunk/tests/lcmap_tiny_cog_2020.tif",
-            # "fits": "https://fits.gsfc.nasa.gov/samples/WFPC2u5780205r_c0fx.fits",
-            "jpg": "https://github.com/rasterio/rasterio/raw/main/tests/data/389225main_sw_1965_1024.jpg",
-        }
-
-        for filetype, url in examples.items():
-            if filetype in ["grib", "jpg", "hdf4"]:
-                with pytest.raises(NotImplementedError):
-                    vds = open_virtual_dataset(url, reader_options={})
-            else:
-                vds = open_virtual_dataset(url, reader_options={})
-                assert isinstance(vds, xr.Dataset)
+            # ("hdf5", "https://github.com/fsspec/kerchunk/raw/main/kerchunk/tests/NEONDSTowerTemperatureData.hdf5"),
+            pytest.param(
+                "tiff",
+                "https://github.com/fsspec/kerchunk/raw/main/kerchunk/tests/lcmap_tiny_cog_2020.tif",
+                marks=pytest.mark.skipif(
+                    not has_tifffile, reason="package tifffile is not available"
+                ),
+            ),
+            pytest.param(
+                "fits",
+                "https://fits.gsfc.nasa.gov/samples/WFPC2u5780205r_c0fx.fits",
+                marks=pytest.mark.skipif(
+                    not has_astropy, reason="package astropy is not available"
+                ),
+            ),
```

Comment:
We could potentially do this same `skipif` thing for dependencies required to read the other filetypes too.